provider/src/main/java/com/cloudera/director/google/sql/GoogleCloudSQLInstanceTemplateConfigurationValidator.java

```diff
@@ -0,0 +1,264 @@
+   * @param localizationContext the localization context
+   */
+  void checkTier(Configured configuration, PluginExceptionConditionAccumulator accumulator,
+      LocalizationContext localizationContext) {
+
+    GoogleCredentials credentials = provider.getCredentials();
+    SQLAdmin sqlAdmin = credentials.getSQLAdmin();
+    String projectId = credentials.getProjectId();
+    String tierName = configuration.getConfigurationValue(TIER, localizationContext);
+
+    try {
+      List<Tier> tierList = sqlAdmin.tiers().list(projectId).execute().getItems();
+
+      if (tierList != null) {
+        for (Tier tier : tierList) {
+          if (tier.getTier().equals(tierName)) {
+            return;
+          }
+        }
+      }
+      addError(accumulator, TIER, localizationContext, null, INVALID_TIER_MSG, tierName);
+    } catch (IOException e) {
+      throw new TransientProviderException(e);
+    }
+  }
+
+  /**
+   * Validates the configured password.
+   *
+   * @param configuration       the configuration to be validated
+   * @param accumulator         the exception condition accumulator
+   * @param localizationContext the localization context
+   */
+  static void checkPassword(Configured configuration, PluginExceptionConditionAccumulator accumulator,
+      LocalizationContext localizationContext) {
+
+    String instancePassword = configuration.getConfigurationValue(MASTER_USER_PASSWORD, localizationContext);
+
+    LOG.info(">> Validating password");
+
+    if (instancePassword == null) {
+      addError(accumulator, instancePassword, localizationContext, null, PASSWORD_MISSING_MSG);
+    } else {
+      int length = instancePassword.length();
+
+      if (length < 1 || length > 16) {
+        addError(accumulator, MASTER_USER_PASSWORD, localizationContext, null, INVALID_PASSWORD_LENGTH_MSG);
+      }
+    }
+  }
+
+  /**
+   * Validates the preferred location.
+   *
+   * @param configuration       the configuration to be validated
+   * @param accumulator         the exception condition accumulator
+   * @param localizationContext the localization context
+   */
+  void checkPreferredLocation(Configured configuration, PluginExceptionConditionAccumulator accumulator,
+      LocalizationContext localizationContext) {
+
+    String preferredLocation = configuration.getConfigurationValue(PREFERRED_LOCATION, localizationContext);
+
+    if (preferredLocation != null && !preferredLocation.isEmpty()) {
+      LOG.info(">> Querying zone '{}'", preferredLocation);
+
+      GoogleCredentials credentials = provider.getCredentials();
+      Compute compute = credentials.getCompute();
+      String projectId = credentials.getProjectId();
+      String regionName = provider.getConfigurationValue(REGION_SQL, localizationContext);
+
+      // US region is called differently in GCE and Google Cloud SQL.
+      String computeRegionName = regionName;
+      if (computeRegionName.equals("us-central")) {
```

Comment:
If you think it will be fixed before you ship a version of the plugin, then I would leave this code in place as a workaround until then and then remove it when you can. If not, then I encourage you to consider something more flexible, like a configurable mapping file.